source/game/field/obj/ObjectShip64.cc

```diff
@@ -0,0 +1,73 @@
+#include "ObjectShip64.hh"
+
+namespace Field {
+
+/// @addr{0x80765C94}
+ObjectShip64::ObjectShip64(const System::MapdataGeoObj &params) : ObjectCollidable(params) {}
+
+/// @addr{0x80765DB0}
+ObjectShip64::~ObjectShip64() = default;
+
+/// @addr{0x80765E30}
+void ObjectShip64::init() {
+    m_railInterpolator->init(0, 0);
+
+    m_tangent = m_railInterpolator->curTangentDir();
+    m_railInterpolator->setPerPointVelocities(true);
+
+    calc();
+    calcModel();
+}
+
+/// @addr{0x80766144}
+void ObjectShip64::calc() {
+    m_railInterpolator->calc();
+
+    setPos(m_railInterpolator->curPos());
+
+    m_tangent = Interpolate(0.2f, m_tangent, m_railInterpolator->curTangentDir());
+    m_tangent.normalise();
+
+    setMatrixFromOrthonormalBasisAndPos(m_tangent);
+}
+
+/// @addr{0x80766864}
+void ObjectShip64::createCollision() {
+    ObjectCollidable::createCollision();
+
+    constexpr f32 RADIUS = 1500.0f;
+    constexpr f32 HEIGHT = 3500.0f;
```

Comment:
Can you move these to the beginning of the function, before the call to `ObjectCollidable::createCollision()`? Sorry this is pedantic, but our convention thus far has been to put `constexpr` variables first before anything else in a function.